scala/BUILD

```diff
@@ -0,0 +1,20 @@
+load("//scala:scala_toolchain.bzl", "scala_toolchain")
+
+#call scala toolchain
+scala_toolchain(
+      name = 'scala_toolchain_impl',
+      scalacopts = [],
+      visibility = ["//visibility:public"]
+    )
+    
+toolchain_type(
+    name = "scala_toolchain_type",
+    visibility = ["//visibility:public"]
+)
+
+toolchain(
+    name = 'scala_toolchain',
```

Comment:
this is confusing to me. Does this string just coincidentally match the load or the item on line 4? Is this a different namespace? Can we possible not have such a name collision to avoid any confusion?

Reply:
You are absolutely right. I see what we can do to avoid it

Reply:
It's just a target name, and doesn't need to match anything, and is referenced from the register_toolchains call in the WORKSPACE.